sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -871,6 +871,17 @@ case class GpuHashAggregateExec(
             }
           }
         }
+        // If cvs is empty, we add a single row with zero value.
+        // This is to fix a bug in the plugin where a paramater-less count wasn't returning the
+        // desired result compared to Spark-CPU.
+        // For more details go to https://github.com/NVIDIA/spark-rapids/issues/1737
+        if (cvs.isEmpty) {
```

Comment:
1. Should we assert that the aggregate function is indeed `count`?
2. Does this only happen with reductions?
3. Need tests for this in the integration tests at least. I'd also expect them in the scala tests... though I defer to @revans2 where he'd like to see new tests go in.

Reply:
So in a count with no parameter what is `agg.ref` above on line 866? It is pointing to something? How is it that we are not getting an NPE above?

Reply:
`aggModeCudfAggregates` is empty so that entire loop is never executed

Reply:
> 

    1. Should we assert that the aggregate function is indeed count?
    2. Does this only happen with reductions?
    3. Need tests for this in the integration tests at least. I'd also expect them in the scala tests... though I defer to @revans2 where he'd like to see new tests go in.

1. We can, I am not sure its needed as no other aggregate is parameterless. @revans2 is this needed? 
2. I am not sure
3. I have added tests

Reply:
I am sorry but the code around this is too complicated for me to be able to really grok it and see if what you have done is correct.  I personally think we need to refactor all of hash aggregate so it is actually understandable without spending a few days to dig into it.  I'll try and find some time to dig into the code, but for now I don't feel comfortable approving this.  I'll let @abellina deal with it.  Thanks for answering my questions.